### ISSUE TITLE
Allow the docker build and docker tag steps of the GitHub Actions CI to be skipped

### ISSUE THREAD

**Issue by combiz** [nf-core/tools#1205]
The `.github/workflows/ci.yml` workflow includes steps to build and tag a docker image: -

```
- name: Build new docker image
  if: env.GIT_DIFF
  run: docker build --no-cache . -t nfcore/example:1.0.0
- name: Pull docker image
  if: ${{ !env.GIT_DIFF }}
  run: |
    docker pull nfcore/example:dev
    docker tag nfcore/example:dev nfcore/example:1.0.0
```

However, it's not currently possible to skip this step with an entry in the `.nf-core.yml` file.  Removing the above steps from the `ci.yml` workflow causes lint errors, e.g.

```
│ actions_ci: CI is not building the correct docker image. Should be: docker build --no-cache . -t almurphy/scfdev:dev                                │
│ actions_ci: CI is not pulling the correct docker image. Should be: docker pull almurphy/scfdev:dev                                                  │
│ actions_ci: CI is not tagging docker image correctly. Should be: docker tag almurphy/scfdev:dev almurphy/scfdev:dev    
```

This is due to this check: "The Docker container for the pipeline must use the correct pipeline version number:" (https://nf-co.re/tools-docs/lint_tests/actions_ci.html).

tl;dr allow docker build / tag to be skipped in the actions_ci workflow.

**Comment by drpatelh** [comment]
We should just delete all of the checks relating to building the Docker container because we will no longer need these for DSL2. Can do a similar thing for any checks related to the Conda environment - `environment.yml`.

**Comment by ewels** [comment]
This has been removed at some point now.
